gpytorch/utils/broadcasting.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import torch
+
+
+def _mul_broadcast_shape(shape_a, shape_b):
```

Comment:
It really hurts me to write this kind of code in python. This should live in C++ in ATen. Ideally, these shape checks would be factored out of the respective tensor operations there, and get an API so it can be called standalone.